app/controllers/products_controller.rb

```diff
@@ -0,0 +1,45 @@
+class ProductsController < ApplicationController
+  before_action :set_nav
+  before_action :set_product, except: [:new, :create]
```

Comment:
So to make the code more parsable later on, would it be better to set @product before all actions? Or to comment the code?